### PR TITLE
fix: Exceptions during parsing Kamelets from OCP

### LINKED ISSUE #656: [BUG] Kaoto is unable to show all Kamelets steps from OCP cluster
**Describe the bug**
Let's say, I have a cluster where Camel-k 1.12.0 (with Apache Camel Kamelets 3.20.1.1) is installed. When I run the Kaoto backend and set correctly OpenShift namespace via `-Dkaoto.openshift.namespace=mkralik -Dkaoto.openshift.catalog-namespace=mkralik`, the Kaoto shows all Kamelets correctly under the Deployments section, however, the Kamelets steps catalog is empty and contains only the Knative step.

**To Reproduce**
1. Install Camel-k 1.12.0 (or 1.10.0) [or Apache Camel Kamelets 3.20.2] on OCP cluster [make sure that that cluster is in the the context in your kubeconfig]
2. Run kaoto-backend
3. Check if all Kamelets steps are loaded ( ~209) in api `/v1/steps?kind=Kamelet`

**Expected behavior**
All Kamelets are shown also in the step catalog for KameletsBindings.

**Environment (please complete the following information):**
 - OS: Fedora 39, OCP 4.12, CRC
 - Kaoto Version: main branch [jar and native], 1.0.0 [native]

**Additional context**
- The whole Apache Camel Kamelets catalog must be on the cluster. When there are only a few Kamelets, it works
- When the kubeconfig is not available, the Kamelets steps are correctly loaded from resources so the issue looks cluster-related.
- Not able to reproduce in dev mode `mvn quarkus:dev -pl api`

### PR DESCRIPTION
**Pull request by mkralik3**
- Fix for https://github.com/KaotoIO/kaoto-backend/issues/656
- Reproducer for https://github.com/KaotoIO/kaoto-backend/issues/656
- Add `AvoidStarImport` rule into checkstyle

### CODE REVIEW

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java [hunk 1]
```diff
@@ -10,21 +12,20 @@
...
 import java.util.Comparator;
 import java.util.List;
 import java.util.Objects;
-import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (Delawen) [inline kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:28]: We try not to compress the imports, for clarity.

**Review by Delawen — APPROVED** [review]

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java (hunk 1 above)
Comment (mkralik3) [reply kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:28]: ok, I will revert it. Also, I can extend main [checkstyle](https://github.com/KaotoIO/kaoto-backend/blob/main/checkstyle.xml) file about it.

**Comment by mkralik3** [comment]
Back to draft in order to fix Github API limits issue in workflow for reproducer

**Comment by mkralik3** [comment]
Updates from previous review:
- Updated workflow and reproducer to fix GitHub api limits
- Add `AvoidStarImport` rule into checkstyle

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java [hunk 2]
```diff
@@ -30,6 +39,11 @@
 @WithKubernetesTestServer
 class KameletParseCatalogTest {
 
+    @ConfigProperty(name = "GITHUB_TOKEN")
+    private Optional<String> githubToken;
```
Comment (Delawen) [inline kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:43]: Uh, does this mean the tests won't run on my local machine?

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java [hunk 3]
```diff
@@ -155,6 +172,36 @@ void testCluster() {
         assertEquals(1, steps.size());
     }
 
+    //reproducer for issue #656
+    @Test
+    void testClusterWithWholeKamelets() throws IOException {
+        int numberOfKamelets = 0;
+
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpGet request =
```
Comment (Delawen) [inline kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:181]: Can't we use the public link instead? https://github.com/apache/camel-kamelets/archive/refs/tags/v3.20.4.zip

**Review by Delawen — CHANGES_REQUESTED** [review]
Just wondering if maybe we can get rid of the github api by acessing either the zip file or the git repository directly.

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java (hunk 2 above)
Comment (Delawen) [reply kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:43]: I guess not, I see a proper handling below.

Comment (mkralik3) [reply kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:43]: No. When the token is not provided and is empty, the Authorization header [is not set](https://github.com/KaotoIO/kaoto-backend/blob/01ff5718fa3ba19d9387025e2d738c30c29587e9/kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java#L183) so if your machine has not exceeded the Github API limits, the test will work.

File: kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java (hunk 3 above)
Comment (mkralik3) [reply kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/kamelet/KameletParseCatalogTest.java:181]: Yes, but after that, the test needs to extract files and loads them via Kubernetes client.

**Comment by mkralik3** [comment]
@Delawen What does it mean `git repository directly.` , clone `org.eclipse.jgit.api.Git` and load files? 

I think we can also use `camel-kamelets-3.20.2.jar` from the resources, I just wanted to avoid working with files in tests.

**Comment by Delawen** [comment]
As it is a public repository, you don't need a token to access it. There's no limit. 

> I just wanted to avoid working with files in tests.

Can I ask why?

**Comment by mkralik3** [comment]
@Delawen Honestly, I found it unnecessary for only one test case when doing an HTTP request is quicker, but yes if we use file, we don't need to take care of github token. So, I am going to rewrite it to use that .jar which is also used by other tests

**Review by Delawen — APPROVED** [review]

**Pull request merged**